### DISCUSSION TITLE
Formatting error when import geometry into 3DEC

### DISCUSSION THREAD

**Discussion by mqc007** [neperfepx/neper#458] [Q&A] [answered]
Hi, I'm using 3DEC to simulate some project. I learnt that with the help with neper can have better tesslation. 
but when I tried to import the stl file into 3DEC as usual, it failed. 
then i tried open the .3dec format file within the 3DEC  it's also failed.
but i can read it with Gmesh. 
just accidental i noticed  when i opened the .stl file exported gmesh remind me that "the file without geo extension "

i think i have done something wrong with the file format 
and i dont know how to creat the tess with Length unit  like mm
i wanna to create a cube with 40mm length 

i tried with -domain (10,10,10)
there is 70000+ cells that's too much 

and here is the code i used in neper 

neper -T -n from_morpho -domain "cube(3,3,3)" -morpho "diameq:custom(1123),1-sphericity:lognormal(0.145,0.03)" -statcell diameq -o mqc333 -format stl

thank you for reading.

**Comment by rquey** [comment, marked as answer]
I don't know for 3dec and stl.

There is no unit length in Neper, so that you can choose it to be millimeters, meters, etc., as long as all inputs are consistent.

**Reply by mqc007** [reply]
Thanks for your answer, i opened the "wrong format" file with gmesh, and there is no point and no volume. only one surface counted. 

There is no entity in my .stl file is that right for neper?  I guess that might be the problem when I tried to import file into 3DEC. 

Is there any possible I can create entity with the file export by neper in Gmesh? 

By the way, would you please give me some suggestions to import neper's file into 3DEC or other softwares. 

Many thanks and best regards.

**Reply by rquey** [reply]
I added the 3dec and stl formats a while ago, for users who provided specifics about them (and they seemed to be happy about the files).
I do not use these formats myself, but they should not have changed...  Sorry I can't help.

**Reply by mqc007** [reply]
Thanks anyway, I've got contact with a researcher who have achieved that method. 

Other post moved here by @rquey:
> I'm back. this is NOT a question, it's a suggestion for people who are new to neper and 3dec. neper can be imported into 3dec version 5.2, with format .3dec. For neper, i highly recommended to open reg 1. for 3dec, u need open item choose data file with edit mode. if u are using 3dec7.0 there is a command conversation choose in edit. and make the "fish safe" off would be better. a suggestion is "DO NOT try other formats".
